ql/src/java/org/apache/hadoop/hive/ql/reexec/ReExecutionStrategyType.java

```diff
@@ -39,6 +37,10 @@ public Class<? extends IReExecutionPlugin> getPluginClass() {
   }
 
   public static Class<? extends IReExecutionPlugin> getPluginClassByName(String strategy) {
-    return ReExecutionStrategyType.valueOf(strategy.toUpperCase()).getPluginClass();
+    try {
+      return ReExecutionStrategyType.valueOf(strategy.toUpperCase()).getPluginClass();
+    } catch (IllegalArgumentException e) {
+      return null;
```

Comment:
no need for that change

Reply:
Yes, with the above changes it is no longer necessary.